features/nanostack/FEATURE_NANOSTACK/nanostack-interface/NanostackInterface.cpp

```diff
@@ -171,6 +165,38 @@ bool NanostackSocket::open(void)
         tr_error("NanostackSocket::open() failed");
         return false;
     }
+
+    if (proto == SOCKET_TCP) {
+        /* Receive and send buffers enabled by default */
+        mode = SOCKET_MODE_OPENED;
+    } else {
+        static const int32_t rcvbuf_size = 2048;
+        socket_setsockopt(temp_socket, SOCKET_SOL_SOCKET, SOCKET_SO_RCVBUF, &rcvbuf_size, sizeof rcvbuf_size);
+        mode = SOCKET_MODE_DATAGRAM;
+    }
+
+    return attach(temp_socket);
+}
+
+int NanostackSocket::accept(NanostackSocket *accepted_socket, ns_address_t *addr)
+{
+    nanostack_assert_locked();
+    MBED_ASSERT(SOCKET_MODE_LISTENING == mode && SOCKET_MODE_UNOPENED == accepted_socket->mode);
+
+    int temp_socket = socket_accept(socket_id, addr, socket_callback);
+    if (temp_socket < 0) {
+        tr_error("NanostackSocket::accept() failed");
+        return temp_socket;
+    }
+    if (!accepted_socket->attach(temp_socket)) {
+        return -1;
+    }
+    accepted_socket->mode = SOCKET_MODE_STREAM;
+    return temp_socket;
+}
+
+bool NanostackSocket::attach(int8_t temp_socket)
+{
```

Comment:
Do we have a test for that? Thought that the "am I locked" test got removed due to portability issues or something.